src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -71,122 +82,242 @@ public SecurityIndexAccessEvaluator(final Settings settings, AuditLog auditLog,
         this.systemIndexMatcher = WildcardMatcher.from(
             settings.getAsList(ConfigConstants.SECURITY_SYSTEM_INDICES_KEY, ConfigConstants.SECURITY_SYSTEM_INDICES_DEFAULT)
         );
-        this.systemIndexEnabled = settings.getAsBoolean(
+        this.protectedSystemIndexMatcher = WildcardMatcher.from(this.securityIndex);
+        this.isSystemIndexEnabled = settings.getAsBoolean(
             ConfigConstants.SECURITY_SYSTEM_INDICES_ENABLED_KEY,
             ConfigConstants.SECURITY_SYSTEM_INDICES_ENABLED_DEFAULT
         );
-
         final boolean restoreSecurityIndexEnabled = settings.getAsBoolean(
             ConfigConstants.SECURITY_UNSUPPORTED_RESTORE_SECURITYINDEX_ENABLED,
             false
         );
 
-        final List<String> securityIndexDeniedActionPatternsList = new ArrayList<String>();
+        final List<String> securityIndexDeniedActionPatternsList = deniedActionPatterns();
+
+        final List<String> securityIndexDeniedActionPatternsListNoSnapshot = new ArrayList<>(securityIndexDeniedActionPatternsList);
```

Comment:
Nit: This is already in a class called SecurityIndexAccessEvaluator so wdyt about shortening these variable names?

```
deniedActionPatternsList
deniedActionPatternsListNoSnapshot
```

Reply:
Want to make a follow up PR to change - I agree.